--- origsrc/webkit-1.1.3/GNUmakefile.am	2009-03-29 14:37:05.562500000 -0500
+++ src/webkit-1.1.3/GNUmakefile.am	2009-03-29 00:35:25.000000000 -0500
@@ -75,6 +75,9 @@ endif
 if OS_WIN32
 no_undefined = -no-undefined
 endif
+if OS_CYGWIN
+no_undefined = -no-undefined
+endif
 
 if OS_GNU
 version_script = -Wl,--version-script,$(srcdir)/autotools/symbols.filter
--- origsrc/webkit-1.1.3/configure.ac	2009-03-29 14:37:05.578125000 -0500
+++ src/webkit-1.1.3/configure.ac	2009-03-29 00:35:25.000000000 -0500
@@ -52,6 +52,9 @@ case "$host" in
      *-*-darwin*)
        os_darwin=yes
        ;;
+     *-*-cygwin*)
+       os_cygwin=yes
+       ;;
 esac
 
 case "$host_os" in
@@ -603,6 +606,7 @@ AM_CONDITIONAL([OS_UNIX],[test "$os_win3
 AM_CONDITIONAL([OS_LINUX],[test "$os_linux" = "yes"])
 AM_CONDITIONAL([OS_GNU],[test "$os_gnu" = "yes"])
 AM_CONDITIONAL([OS_FREEBSD],[test "$os_freebsd" = "yes"])
+AM_CONDITIONAL([OS_CYGWIN],[test "$os_cygwin" = "yes"])
 
 # target conditionals
 AM_CONDITIONAL([TARGET_X11], [test "$with_target" = "x11"])
