--- origsrc/webkit-1.1.10/JavaScriptCore/API/JSStringRef.h	2008-06-17 13:17:45.000000000 -0500
+++ src/webkit-1.1.10/JavaScriptCore/API/JSStringRef.h	2009-06-17 22:26:34.748257000 -0500
@@ -37,7 +37,7 @@
 extern "C" {
 #endif
 
-#if !defined(WIN32) && !defined(_WIN32)
+#if !defined(WIN32) && !defined(_WIN32) && !defined(__CYGWIN__)
 /*!
 @typedef JSChar
 @abstract A Unicode character.
--- origsrc/webkit-1.1.10/JavaScriptCore/wtf/DateMath.h	2009-06-09 07:10:42.000000000 -0500
+++ src/webkit-1.1.10/JavaScriptCore/wtf/DateMath.h	2009-06-17 22:26:36.094527600 -0500
@@ -109,7 +109,7 @@ struct GregorianDateTime : Noncopyable {
         , year(inTm.tm_year)
         , isDST(inTm.tm_isdst)
     {
-#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !COMPILER(RVCT)
+#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !PLATFORM(CYGWIN) && !COMPILER(RVCT)
         utcOffset = static_cast<int>(inTm.tm_gmtoff);
 
         int inZoneSize = strlen(inTm.tm_zone) + 1;
@@ -136,7 +136,7 @@ struct GregorianDateTime : Noncopyable {
         ret.tm_year  =  year;
         ret.tm_isdst =  isDST;
 
-#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !COMPILER(RVCT)
+#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !PLATFORM(CYGWIN) && !COMPILER(RVCT)
         ret.tm_gmtoff = static_cast<long>(utcOffset);
         ret.tm_zone = timeZone;
 #endif
--- origsrc/webkit-1.1.10/JavaScriptCore/wtf/Platform.h	2009-06-09 07:10:42.000000000 -0500
+++ src/webkit-1.1.10/JavaScriptCore/wtf/Platform.h	2009-06-17 22:26:36.107027800 -0500
@@ -49,6 +49,13 @@
 #include <TargetConditionals.h>
 #endif
 
+/* PLATFORM(CYGWIN) */
+/* Operating system level dependencies for Cygwin that should */
+/* be used regardless of operating environment */
+#ifdef __CYGWIN__
+#define WTF_PLATFORM_CYGWIN 1
+#endif
+
 /* PLATFORM(WIN_OS) */
 /* Operating system level dependencies for Windows that should be used */
 /* regardless of operating environment */
@@ -113,6 +120,7 @@
 /* Operating system level dependencies for Unix-like systems that */
 /* should be used regardless of operating environment */
 #if   PLATFORM(DARWIN)     \
+   || PLATFORM(CYGWIN)     \
    || PLATFORM(FREEBSD)    \
    || PLATFORM(S60)        \
    || PLATFORM(NETBSD)     \
--- origsrc/webkit-1.1.10/WebCore/plugins/PluginDatabase.cpp	2008-11-01 14:12:05.000000000 -0500
+++ src/webkit-1.1.10/WebCore/plugins/PluginDatabase.cpp	2009-06-17 22:26:36.112027900 -0500
@@ -351,7 +351,9 @@ void PluginDatabase::getPluginPathsInDir
     // FIXME: This should be a case insensitive set.
     HashSet<String> uniqueFilenames;
 
-#if defined(XP_UNIX)
+#if defined(__CYGWIN__)
+    String fileNameFilter("*.dll");
+#elif defined(XP_UNIX)
     String fileNameFilter("*.so");
 #else
     String fileNameFilter("");
--- origsrc/webkit-1.1.10/WebKitTools/DumpRenderTree/DumpRenderTree.h	2009-05-13 01:45:12.000000000 -0500
+++ src/webkit-1.1.10/WebKitTools/DumpRenderTree/DumpRenderTree.h	2009-06-17 22:26:36.117028000 -0500
@@ -46,7 +46,7 @@
 
 #include <string>
 
-#if !PLATFORM(OPENBSD)
+#if !PLATFORM(OPENBSD) && !PLATFORM(CYGWIN)
 std::wstring urlSuitableForTestResult(const std::wstring& url);
 #endif
 
