--- origsrc/webkit-1.1.15.3/JavaScriptCore/icu/unicode/platform.h	2009-09-22 10:29:13.000000000 -0500
+++ src/webkit-1.1.15.3/JavaScriptCore/icu/unicode/platform.h	2009-10-28 12:16:04.039842200 -0500
@@ -210,7 +210,11 @@ typedef unsigned int uint32_t;
 /*===========================================================================*/
 
 #define U_HAVE_WCHAR_H      1
+#if defined(__CYGWIN__)
+#define U_SIZEOF_WCHAR_T    2
+#else
 #define U_SIZEOF_WCHAR_T    4
+#endif
 
 #define U_HAVE_WCSCPY       1
 
--- origsrc/webkit-1.1.15.3/JavaScriptCore/icu/unicode/umachine.h	2009-09-22 10:29:13.000000000 -0500
+++ src/webkit-1.1.15.3/JavaScriptCore/icu/unicode/umachine.h	2009-10-28 12:12:50.973905600 -0500
@@ -298,7 +298,7 @@ typedef int8_t UBool;
 #       if (U_SIZEOF_WCHAR_T==4)
 #           define U_WCHAR_IS_UTF32
 #       endif
-#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64) || defined(__CYGWIN__)
 #       define U_WCHAR_IS_UTF16    
 #   endif
 #endif
--- origsrc/webkit-1.1.15.3/WebCore/icu/unicode/platform.h	2009-09-22 10:29:21.000000000 -0500
+++ src/webkit-1.1.15.3/WebCore/icu/unicode/platform.h	2009-10-28 13:03:48.572062600 -0500
@@ -210,7 +210,11 @@ typedef unsigned int uint32_t;
 /*===========================================================================*/
 
 #define U_HAVE_WCHAR_H      1
+#if defined(__CYGWIN__)
+#define U_SIZEOF_WCHAR_T    2
+#else
 #define U_SIZEOF_WCHAR_T    4
+#endif
 
 #define U_HAVE_WCSCPY       1
 
--- origsrc/webkit-1.1.15.3/WebCore/icu/unicode/umachine.h	2009-09-22 10:29:21.000000000 -0500
+++ src/webkit-1.1.15.3/WebCore/icu/unicode/umachine.h	2009-10-28 13:04:33.968141800 -0500
@@ -298,7 +298,7 @@ typedef int8_t UBool;
 #       if (U_SIZEOF_WCHAR_T==4)
 #           define U_WCHAR_IS_UTF32
 #       endif
-#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64) || defined(__CYGWIN__)
 #       define U_WCHAR_IS_UTF16    
 #   endif
 #endif
--- origsrc/webkit-1.1.15.3/WebKit/gtk/WebCoreSupport/EditorClientGtk.cpp	2009-10-20 16:18:44.000000000 -0500
+++ src/webkit-1.1.15.3/WebKit/gtk/WebCoreSupport/EditorClientGtk.cpp	2009-11-03 21:11:16.544456900 -0600
@@ -609,7 +609,7 @@ void EditorClient::checkSpellingOfString
     if (!langs)
         return;
 
-    gchar* ctext = g_utf16_to_utf8(const_cast<gunichar2*>(text), length, 0, 0, 0);
+    gchar* ctext = g_utf16_to_utf8(reinterpret_cast<const gunichar2*>(text), length, 0, 0, 0);
     int utflen = g_utf8_strlen(ctext, -1);
 
     PangoLanguage* language = pango_language_get_default();
