--- origsrc/webkit-1.1.15.3/WebKitTools/GNUmakefile.am	2009-09-29 14:23:22.000000000 -0500
+++ src/webkit-1.1.15.3/WebKitTools/GNUmakefile.am	2009-11-04 00:32:15.535648500 -0600
@@ -81,8 +81,8 @@ Programs_DumpRenderTree_CFLAGS = \
 	$(LIBSOUP_CFLAGS)
 
 Programs_DumpRenderTree_LDADD = \
-	libwebkit-1.0.la \
 	libJavaScriptCore.la \
+	libwebkit-1.0.la \
 	$(GLOBALDEPS_LIBS) \
 	$(CAIRO_LIBS) \
 	$(GTK_LIBS) \
