--- origsrc/webkitgtk-1.10.2/Source/WebKit2/GNUmakefile.am	2012-12-08 10:50:44.000000000 -0600
+++ src/webkitgtk-1.10.2/Source/WebKit2/GNUmakefile.am	2012-12-18 19:31:31.387765300 -0600
@@ -268,7 +268,7 @@ BUILT_SOURCES += generate-webkit2-forwar
 
 $(GENSOURCES_WEBKIT2)/webkit2gtk/include/webkit2: $(libwebkit2gtkinclude_HEADERS)
 	$(AM_V_GEN)mkdir -p $(GENSOURCES_WEBKIT2)/webkit2gtk/include \
-	&& ln -n -s -f ${shell pwd}/$(WebKit2)/UIProcess/API/gtk $@
+	&& ln -n -s -f $(WebKit2)/UIProcess/API/gtk $@
 
 BUILT_SOURCES += $(GENSOURCES_WEBKIT2)/webkit2gtk/include/webkit2
 
