--- origsrc/webkit-1.2.1/GNUmakefile.am	2010-06-15 18:46:42.643885700 -0500
+++ src/webkit-1.2.1/GNUmakefile.am	2010-06-15 19:06:37.672237400 -0500
@@ -420,7 +420,7 @@ $(WEBKIT_GIRSOURCES): $(JSCORE_GIRSOURCE
 	     --include=Gtk-2.0 \
 	     --include=JSCore-1.0 \
 	     --include=Soup-2.4 \
-	     --library=webkit-1.0 \
+	     --library=libwebkit-1.0.la \
 	     --libtool="$(LIBTOOL)" \
 	     --pkg gobject-2.0 \
 	     --pkg gtk+-2.0 \
