--- origsrc/webkit-1.2.1/configure.ac	2010-06-16 01:50:11.148169000 -0500
+++ src/webkit-1.2.1/configure.ac	2010-06-15 18:46:42.000000000 -0500
@@ -66,6 +66,9 @@ case "$host" in
      *-*-darwin*)
        os_darwin=yes
        ;;
+     *-*-cygwin*)
+       os_cygwin=yes
+       ;;
 esac
 
 case "$host_os" in
@@ -774,6 +777,7 @@ AM_CONDITIONAL([OS_UNIX],[test "$os_win3
 AM_CONDITIONAL([OS_LINUX],[test "$os_linux" = "yes"])
 AM_CONDITIONAL([OS_GNU],[test "$os_gnu" = "yes"])
 AM_CONDITIONAL([OS_FREEBSD],[test "$os_freebsd" = "yes"])
+AM_CONDITIONAL([OS_CYGWIN],[test "$os_cygwin" = "yes"])
 
 # target conditionals
 AM_CONDITIONAL([TARGET_X11], [test "$with_target" = "x11"])
--- origsrc/webkit-1.2.1/GNUmakefile.am	2010-06-16 01:50:11.141168600 -0500
+++ src/webkit-1.2.1/GNUmakefile.am	2010-06-15 19:06:37.672237400 -0500
@@ -124,6 +124,9 @@ corekit_cppflags += \
 if !TARGET_WIN32
 corekit_cppflags += -DXP_UNIX
 endif
+if OS_CYGWIN
+no_undefined = -no-undefined
+endif
 
 # Default compiler flags
 global_cflags += \
--- origsrc/webkit-1.2.1/JavaScriptCore/API/JSStringRef.h	2010-05-07 09:09:08.000000000 -0500
+++ src/webkit-1.2.1/JavaScriptCore/API/JSStringRef.h	2010-06-15 20:03:24.765112000 -0500
@@ -38,6 +38,7 @@ extern "C" {
 #endif
 
 #if !defined(WIN32) && !defined(_WIN32) && !defined(__WINSCW__) \
+    && !defined(__CYGWIN__) \
     && !(defined(__CC_ARM) || defined(__ARMCC__)) /* RVCT */
 /*!
 @typedef JSChar
--- origsrc/webkit-1.2.1/JavaScriptCore/jit/JITStubs.cpp	2010-05-07 09:09:08.000000000 -0500
+++ src/webkit-1.2.1/JavaScriptCore/jit/JITStubs.cpp	2010-06-15 18:58:32.000458600 -0500
@@ -65,7 +65,7 @@ using namespace std;
 
 namespace JSC {
 
-#if OS(DARWIN) || OS(WINDOWS)
+#if OS(CYGWIN) || OS(DARWIN) || OS(WINDOWS)
 #define SYMBOL_STRING(name) "_" #name
 #else
 #define SYMBOL_STRING(name) #name
--- origsrc/webkit-1.2.1/JavaScriptCore/wtf/Platform.h	2010-05-07 09:09:08.000000000 -0500
+++ src/webkit-1.2.1/JavaScriptCore/wtf/Platform.h	2010-06-15 19:18:14.496093500 -0500
@@ -340,6 +340,11 @@
 
 #endif
 
+/* OS(CYGWIN) - Cygwin */
+#ifdef __CYGWIN__
+#define WTF_OS_CYGWIN 1
+#endif
+
 /* OS(IPHONE_OS) - iPhone OS */
 /* OS(MAC_OS_X) - Mac OS X (not including iPhone OS) */
 #if OS(DARWIN) && ((defined(TARGET_OS_EMBEDDED) && TARGET_OS_EMBEDDED)  \
@@ -406,6 +411,7 @@
 /* OS(UNIX) - Any Unix-like system */
 #if   OS(AIX)              \
     || OS(ANDROID)          \
+    || OS(CYGWIN)           \
     || OS(DARWIN)           \
     || OS(FREEBSD)          \
     || OS(HAIKU)            \
@@ -674,7 +680,7 @@
 #define HAVE_SIGNAL_H 1
 #endif
 
-#if !OS(WINDOWS) && !OS(SOLARIS) && !OS(QNX) \
+#if !OS(WINDOWS) && !OS(SOLARIS) && !OS(QNX) && !OS(CYGWIN) \
     && !OS(SYMBIAN) && !OS(HAIKU) && !OS(RVCT) \
     && !OS(ANDROID) && !PLATFORM(BREWMP)
 #define HAVE_TM_GMTOFF 1
@@ -876,9 +882,9 @@
 #define WTF_USE_JSVALUE64 1
 #elif CPU(ARM) || CPU(PPC64) || CPU(MIPS)
 #define WTF_USE_JSVALUE32 1
-#elif OS(WINDOWS) && COMPILER(MINGW)
+#elif OS(CYGWIN) || (OS(WINDOWS) && COMPILER(MINGW))
 /* Using JSVALUE32_64 causes padding/alignement issues for JITStubArg
-on MinGW. See https://bugs.webkit.org/show_bug.cgi?id=29268 */
+on Cygwin/MinGW. See https://bugs.webkit.org/show_bug.cgi?id=29268 */
 #define WTF_USE_JSVALUE32 1
 #else
 #define WTF_USE_JSVALUE32_64 1
--- origsrc/webkit-1.2.1/WebCore/plugins/PluginDatabase.cpp	2010-03-19 10:20:54.000000000 -0500
+++ src/webkit-1.2.1/WebCore/plugins/PluginDatabase.cpp	2010-06-15 18:54:10.589506700 -0500
@@ -410,7 +410,9 @@ void PluginDatabase::getPluginPathsInDir
     // FIXME: This should be a case insensitive set.
     HashSet<String> uniqueFilenames;
 
-#if defined(XP_UNIX)
+#if defined(__CYGWIN__)
+    String fileNameFilter("*.dll");
+#elif defined(XP_UNIX)
     String fileNameFilter("*.so");
 #else
     String fileNameFilter("");
