--- origsrc/webkit-1.2.1/GNUmakefile.am	2010-06-15 18:46:42.643885700 -0500
+++ src/webkit-1.2.1/GNUmakefile.am	2010-06-15 19:06:37.672237400 -0500
@@ -533,7 +533,7 @@ EXTRA_DIST += \
 # extra resource files
 resourcesdir = ${datadir}/webkit-1.0/resources
 dist_resources_DATA = \
-	$(shell ls $(srcdir)/WebKit/gtk/resources/*.html)
+	$(wildcard $(srcdir)/WebKit/gtk/resources/*.html)
 
 # END WEBKIT GTK+
 
--- origsrc/webkit-1.2.4/WebCore/GNUmakefile.am
+++ src/webkit-1.2.4/WebCore/GNUmakefile.am
@@ -3772,14 +3772,14 @@ webcore_dist += \
 webinspectordir = ${datadir}/webkit-1.0/webinspector
 dist_webinspector_DATA = \
 	$(WebCore)/English.lproj/localizedStrings.js \
-	$(shell ls $(WebCore)/inspector/front-end/*.js) \
-	$(shell ls $(WebCore)/inspector/front-end/*.html) \
-	$(shell ls $(WebCore)/inspector/front-end/*.css)
+	$(wildcard $(WebCore)/inspector/front-end/*.js) \
+	$(wildcard $(WebCore)/inspector/front-end/*.html) \
+	$(wildcard $(WebCore)/inspector/front-end/*.css)
 
 webinspectorimagesdir = ${datadir}/webkit-1.0/webinspector/Images
 dist_webinspectorimages_DATA = \
-	$(shell ls $(WebCore)/inspector/front-end/Images/*.gif) \
-	$(shell ls $(WebCore)/inspector/front-end/Images/*.png)
+	$(wildcard $(WebCore)/inspector/front-end/Images/*.gif) \
+	$(wildcard $(WebCore)/inspector/front-end/Images/*.png)
 
 webresourcesdir = ${datadir}/webkit-1.0/images
 dist_webresources_DATA = \
--- origsrc/webkit-1.2.1/WebKit/gtk/po/GNUmakefile.am	2010-03-16 16:55:49.000000000 -0500
+++ src/webkit-1.2.1/WebKit/gtk/po/GNUmakefile.am	2010-06-15 18:55:48.321096700 -0500
@@ -25,7 +25,7 @@ MSGMERGE := $(shell which msgmerge)
 
 MSGMERGE_UPDATE := $(MSGMERGE) --update
 
-POFILES += $(shell ls $(srcdir)/WebKit/gtk/po/*.po)
+POFILES += $(wildcard $(srcdir)/WebKit/gtk/po/*.po)
 
 MOFILES += $(shell echo $(POFILES) | tr ' ' '\n' | sed "s,^$(srcdir)/,,g" | sed 's/\.po/.mo/g')
 
