--- origsrc/webkit-1.2.1/JavaScriptCore/runtime/Collector.cpp	2010-05-07 09:09:08.000000000 -0500
+++ src/webkit-1.2.1/JavaScriptCore/runtime/Collector.cpp	2010-06-15 19:04:30.771979100 -0500
@@ -75,6 +75,11 @@
 #endif
 #include <unistd.h>
 
+#if OS(CYGWIN)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
+
 #if OS(SOLARIS)
 #include <thread.h>
 #else
@@ -574,7 +579,7 @@ static inline void* currentThreadStackBa
         MOV pTib, EAX
     }
     return static_cast<void*>(pTib->StackBase);
-#elif OS(WINDOWS) && CPU(X86) && COMPILER(GCC)
+#elif (OS(CYGWIN) || OS(WINDOWS)) && CPU(X86) && COMPILER(GCC)
     // offset 0x18 from the FS segment register gives a pointer to
     // the thread information block for the current thread
     NT_TIB* pTib;
