--- origsrc/webkit-1.3.12/Source/JavaScriptCore/API/JSStringRef.h	2011-01-05 07:05:14.000000000 -0600
+++ src/webkit-1.3.12/Source/JavaScriptCore/API/JSStringRef.h	2011-03-06 17:06:59.774951800 -0600
@@ -37,7 +37,7 @@
 extern "C" {
 #endif
 
-#if !defined(WIN32) && !defined(_WIN32) && !defined(__WINSCW__) \
+#if !defined(WIN32) && !defined(_WIN32) && !defined(__WINSCW__) && !defined(__CYGWIN__) \
     && !((defined(__CC_ARM) || defined(__ARMCC__)) && !defined(__linux__)) /* RVCT */
 /*!
 @typedef JSChar
--- origsrc/webkit-1.3.12/Source/JavaScriptCore/jit/JITStubs.cpp	2011-02-21 04:18:53.000000000 -0600
+++ src/webkit-1.3.12/Source/JavaScriptCore/jit/JITStubs.cpp	2011-03-06 17:07:22.488591400 -0600
@@ -66,7 +66,7 @@ using namespace std;
 
 namespace JSC {
 
-#if OS(DARWIN) || OS(WINDOWS)
+#if OS(CYGWIN) || OS(DARWIN) || OS(WINDOWS)
 #define SYMBOL_STRING(name) "_" #name
 #else
 #define SYMBOL_STRING(name) #name
@@ -80,7 +80,7 @@ namespace JSC {
 
 #if (OS(LINUX) || OS(FREEBSD)) && CPU(X86_64)
 #define SYMBOL_STRING_RELOCATION(name) #name "@plt"
-#elif OS(DARWIN)
+#elif OS(CYGWIN) || OS(DARWIN)
 #define SYMBOL_STRING_RELOCATION(name) "_" #name
 #elif CPU(X86) && COMPILER(MINGW)
 #define SYMBOL_STRING_RELOCATION(name) "@" #name "@4"
--- origsrc/webkit-1.3.12/Source/JavaScriptCore/jit/JITStubs.h	2011-01-12 08:40:49.000000000 -0600
+++ src/webkit-1.3.12/Source/JavaScriptCore/jit/JITStubs.h	2011-03-06 22:15:39.965070000 -0600
@@ -113,10 +113,10 @@ namespace JSC {
         ReturnAddressPtr* returnAddressSlot() { return reinterpret_cast<ReturnAddressPtr*>(this) - 1; }
     };
 #elif CPU(X86)
-#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
 #pragma pack(push)
 #pragma pack(4)
-#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
     struct JITStackFrame {
         void* reserved; // Unused
         JITStubArg args[6];
@@ -140,9 +140,9 @@ namespace JSC {
         // When JIT code makes a call, it pushes its return address just below the rest of the stack.
         ReturnAddressPtr* returnAddressSlot() { return reinterpret_cast<ReturnAddressPtr*>(this) - 1; }
     };
-#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
 #pragma pack(pop)
-#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
 #elif CPU(ARM_THUMB2)
     struct JITStackFrame {
         JITStubArg reserved; // Unused
--- origsrc/webkit-1.3.12/Source/JavaScriptCore/wtf/Platform.h	2011-02-22 05:11:16.000000000 -0600
+++ src/webkit-1.3.12/Source/JavaScriptCore/wtf/Platform.h	2011-03-06 17:08:40.504327400 -0600
@@ -364,6 +364,11 @@
 #define WTF_OS_AIX 1
 #endif
 
+/* OS(CYGWIN) - Cygwin */
+#ifdef __CYGWIN__
+#define WTF_OS_CYGWIN 1
+#endif
+
 /* OS(DARWIN) - Any Darwin-based OS, including Mac OS X and iPhone OS */
 #ifdef __APPLE__
 #define WTF_OS_DARWIN 1
@@ -451,6 +456,7 @@
 /* OS(UNIX) - Any Unix-like system */
 #if   OS(AIX)              \
     || OS(ANDROID)          \
+    || OS(CYGWIN)           \
     || OS(DARWIN)           \
     || OS(FREEBSD)          \
     || OS(HAIKU)            \
@@ -754,7 +760,7 @@
 
 #if !OS(WINDOWS) && !OS(SOLARIS) && !OS(QNX) \
     && !OS(SYMBIAN) && !OS(HAIKU) && !OS(RVCT) \
-    && !OS(ANDROID) && !PLATFORM(BREWMP)
+    && !OS(ANDROID) && !PLATFORM(BREWMP) && !OS(CYGWIN)
 #define HAVE_TM_GMTOFF 1
 #define HAVE_TM_ZONE 1
 #define HAVE_TIMEGM 1
--- origsrc/webkit-1.3.12/Source/WebCore/plugins/PluginDatabase.cpp	2011-01-12 08:40:53.000000000 -0600
+++ src/webkit-1.3.12/Source/WebCore/plugins/PluginDatabase.cpp	2011-03-06 17:09:35.728423700 -0600
@@ -445,7 +445,9 @@ void PluginDatabase::getPluginPathsInDir
     // FIXME: This should be a case insensitive set.
     HashSet<String> uniqueFilenames;
 
-#if defined(XP_UNIX)
+#if defined(__CYGWIN__)
+    String fileNameFilter("*.dll");
+#elif defined(XP_UNIX)
     String fileNameFilter("*.so");
 #else
     String fileNameFilter("");
--- origsrc/webkit-1.3.12/configure.ac	2011-02-22 12:28:52.000000000 -0600
+++ src/webkit-1.3.12/configure.ac	2011-03-06 17:10:33.464124400 -0600
@@ -47,10 +47,16 @@ m4_ifdef([AM_SILENT_RULES], [AM_SILENT_R
 # TODO: move these to webkit.m4?
 AC_MSG_CHECKING([for native Win32])
 case "$host" in
+     *-*-cygwin*)
+       platform_win32=yes
+       os_win32=no
+       ;;
      *-*-mingw*)
+       platform_win32=yes
        os_win32=yes
        ;;
      *)
+       platform_win32=no
        os_win32=no
        ;;
 esac
@@ -974,6 +980,7 @@ fi
 GTK_DOC_CHECK([1.10])
 
 # OS conditionals
+AM_CONDITIONAL([PLATFORM_WIN32],[test "$platform_win32" = "yes"])
 AM_CONDITIONAL([OS_WIN32],[test "$os_win32" = "yes"])
 AM_CONDITIONAL([OS_UNIX],[test "$os_win32" = "no"])
 AM_CONDITIONAL([OS_LINUX],[test "$os_linux" = "yes"])
