--- origsrc/webkit-1.8.1/Tools/GNUmakefile.am	2012-04-23 20:35:37.000000000 -0500
+++ src/webkit-1.8.1/Tools/GNUmakefile.am	2012-05-13 21:09:48.370543000 -0500
@@ -157,11 +157,11 @@ Programs_DumpRenderTree_CFLAGS = \
 	$(LIBSOUP_CFLAGS)
 
 Programs_DumpRenderTree_LDADD = \
+	libWebCorePlatform.la \
+	libWebCoreModules.la \
+	libWebCoreInternals.la \
 	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
 	libwebkitgtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
-	libWebCorePlatform.la \
-	libWebCoreModules.la \
-	libWebCoreInternals.la \
 	$(GLOBALDEPS_LIBS) \
 	$(CAIRO_LIBS) \
 	$(GTK_LIBS) \
@@ -199,6 +199,7 @@ TestNetscapePlugin_libtestnetscapeplugin
 # For the Gtk port we want to use XP_UNIX both on X11 and Mac
 if !TARGET_WIN32
 TestNetscapePlugin_libtestnetscapeplugin_la_CPPFLAGS += -DXP_UNIX
+TestNetscapePlugin_libtestnetscapeplugin_la_LIBADD = -lX11
 endif
 
 TestNetscapePlugin_libtestnetscapeplugin_la_SOURCES = \
