--- origsrc/webkitgtk-2.0.4/Source/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp	2013-02-12 10:22:45.000000000 -0600
+++ src/webkitgtk-2.0.4/Source/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp	2014-07-04 18:49:03.842613600 -0500
@@ -39,8 +39,9 @@
 #include "TextEncoding.h"
 #include <cairo-ft.h>
 #include <cairo.h>
-#include <freetype/freetype.h>
-#include <freetype/tttables.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_TRUETYPE_TABLES_H
 #include <hb.h>
 #include <wtf/text/CString.h>
 
