https://bugs.webkit.org/show_bug.cgi?id=119798

--- origsrc/webkitgtk-2.0.4/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2013-07-19 02:03:53.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2015-01-02 11:23:25.908484700 -0600
@@ -47,6 +47,7 @@ WHICH GENERATES THE GLSL ES PARSER (glsl
 %expect 1 /* One shift reduce conflict because of if | else */
 %pure-parser
 %parse-param {TParseContext* context}
+%lex-param {YYLEX_PARAM}
 
 %union {
     struct {
