--- origsrc/webkitgtk-2.0.4/Source/WebKit2/GNUmakefile.am	2013-07-19 02:04:18.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/WebKit2/GNUmakefile.am	2014-07-04 15:53:20.643297300 -0500
@@ -280,11 +280,11 @@ BUILT_SOURCES += generate-webkit2-forwar
 
 $(GENSOURCES_WEBKIT2)/webkit2gtk/include/webkit2: $(libwebkit2gtkinclude_HEADERS)
 	$(AM_V_GEN)mkdir -p $(GENSOURCES_WEBKIT2)/webkit2gtk/include \
-	&& ln -n -s -f ${shell pwd}/$(WebKit2)/UIProcess/API/gtk $@
+	&& ln -n -s -f $(WebKit2)/UIProcess/API/gtk $@
 
 $(GENSOURCES_WEBKIT2)/webkit2extension/include/webkit2: $(libwebkit2gtkinclude_HEADERS)
 	$(AM_V_GEN)mkdir -p $(GENSOURCES_WEBKIT2)/webkit2extension/include \
-	&& ln -n -s -f ${shell pwd}/$(WebKit2)/WebProcess/InjectedBundle/API/gtk $@
+	&& ln -n -s -f $(WebKit2)/WebProcess/InjectedBundle/API/gtk $@
 
 BUILT_SOURCES += \
 	$(GENSOURCES_WEBKIT2)/webkit2extension/include/webkit2 \
