--- origsrc/webkitgtk-2.0.4/Source/WebKit/gtk/GNUmakefile.am	2013-07-19 02:04:17.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/WebKit/gtk/GNUmakefile.am	2014-07-04 15:53:20.634296200 -0500
@@ -303,8 +303,8 @@ $(WEBKIT_GIRSOURCES): $(G_IR_SCANNER) $(
 	     --include=Gtk-@GTK_API_VERSION@ \
 	     --include=JSCore-@WEBKITGTK_API_VERSION@ \
 	     --include=Soup-2.4 \
-	     --library=webkitgtk-@WEBKITGTK_API_VERSION@ \
-	     --library=javascriptcoregtk-@WEBKITGTK_API_VERSION@ \
+	     --library=libwebkitgtk-@WEBKITGTK_API_VERSION@.la \
+	     --library=libjavascriptcoregtk-@WEBKITGTK_API_VERSION@.la \
 	     --libtool="$(LIBTOOL)" \
 	     --pkg=gobject-2.0 \
 	     --pkg=gtk+-@GTK_API_VERSION@ \
--- origsrc/webkitgtk-2.0.4/Source/WebKit/gtk/JSCore.gir.in	2013-07-19 02:04:17.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/WebKit/gtk/JSCore.gir.in	2014-07-04 15:53:20.637796600 -0500
@@ -2,7 +2,7 @@
 <repository version="1.2"
             xmlns="http://www.gtk.org/introspection/core/1.0"
             xmlns:c="http://www.gtk.org/introspection/c/1.0">
-  <namespace name="JSCore" version="@WEBKITGTK_API_VERSION@" shared-library="webkitgtk-@WEBKITGTK_API_VERSION@">
+  <namespace name="JSCore" version="@WEBKITGTK_API_VERSION@" shared-library="cygjavascriptcoregtk-@WEBKITGTK_API_VERSION@-0.dll">
     <alias name="GlobalContextRef" target="none">
       <type name="JSGlobalContextRef" c:type="JSGlobalContextRef"/>
     </alias>
--- origsrc/webkitgtk-2.0.4/Source/WebKit2/GNUmakefile.am	2013-07-19 02:04:18.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/WebKit2/GNUmakefile.am	2014-07-04 15:53:20.643297300 -0500
@@ -228,8 +228,8 @@ WebKit2-@WEBKITGTK_API_VERSION@.gir: $(G
 	     --include=GObject-2.0 \
 	     --include=Gtk-3.0 \
 	     --include=Soup-2.4 \
-	     --library=webkit2gtk-@WEBKITGTK_API_VERSION@ \
-	     --library=javascriptcoregtk-@WEBKITGTK_API_VERSION@ \
+	     --library=libwebkit2gtk-@WEBKITGTK_API_VERSION@.la \
+	     --library=libjavascriptcoregtk-@WEBKITGTK_API_VERSION@.la \
 	     --libtool="$(LIBTOOL)" \
 	     --pkg=gobject-2.0 \
 	     --pkg=gtk+-@GTK_API_VERSION@ \
