--- origsrc/webkitgtk-2.0.4/GNUmakefile.am	2013-07-19 01:58:38.000000000 -0500
+++ src/webkitgtk-2.0.4/GNUmakefile.am	2014-07-04 15:53:20.559286600 -0500
@@ -155,9 +155,9 @@ feature_defines := $(shell cat WebKitFea
 
 # -no-undefined required for building DLLs on Windows
 # It breaks the build on other platforms, so we use it conditionally
-if OS_WIN32
+if PLATFORM_WIN32
 no_undefined = -no-undefined
-version_script = -export-symbols-regex "^(webkit_|k?JS).*"
+version_script = -export-symbols-regex "^(webkit_|k?JS|webkitWebSettingsUserAgentForURI).*"
 endif
 
 if OS_GNU
--- origsrc/webkitgtk-2.0.4/Source/WebCore/GNUmakefile.am	2013-07-19 02:03:56.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/WebCore/GNUmakefile.am	2014-07-04 15:53:20.620294400 -0500
@@ -463,6 +463,7 @@ libWebCore_la_SOURCES = \
 	$(webcore_sources)
 
 libWebCore_la_CXXFLAGS = \
+	-g1 \
 	-fvisibility-inlines-hidden \
 	$(global_cxxflags)
 
--- origsrc/webkitgtk-2.0.4/Source/autotools/CheckSystemAndBasicDependencies.m4	2013-07-19 02:04:20.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/autotools/CheckSystemAndBasicDependencies.m4	2014-07-04 15:53:20.649298100 -0500
@@ -2,15 +2,20 @@ AC_CANONICAL_HOST
 AC_MSG_CHECKING([for native Win32])
 case "$host" in
      *-*-mingw*)
+       platform_win32=yes
        os_win32=yes
        ;;
      *)
+       platform_win32=no
        os_win32=no
        ;;
 esac
 AC_MSG_RESULT([$os_win32])
 
 case "$host" in
+     *-*-cygwin*)
+       platform_win32=yes
+       ;;
      *-*-linux*)
        os_linux=yes
        ;;
--- origsrc/webkitgtk-2.0.4/Source/autotools/SetupAutomake.m4	2013-07-19 02:04:21.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/autotools/SetupAutomake.m4	2014-07-04 15:53:20.652298400 -0500
@@ -15,6 +15,7 @@ AM_CONDITIONAL([GTK_API_VERSION_2],[test
 AM_WEBKIT_FEATURE_CONDITIONAL([ENABLE_INDEXED_DATABASE])
 
 # OS conditionals.
+AM_CONDITIONAL([PLATFORM_WIN32],[test "$platform_win32" = "yes"])
 AM_CONDITIONAL([OS_WIN32],[test "$os_win32" = "yes"])
 AM_CONDITIONAL([OS_UNIX],[test "$os_win32" = "no"])
 AM_CONDITIONAL([OS_LINUX],[test "$os_linux" = "yes"])
--- origsrc/webkitgtk-2.0.4/Source/autotools/SetupCompilerFlags.m4	2013-07-19 02:04:21.000000000 -0500
+++ src/webkitgtk-2.0.4/Source/autotools/SetupCompilerFlags.m4	2014-07-04 15:53:20.655298800 -0500
@@ -1,5 +1,5 @@
 # Use C99 as the language standard for C code.
-CFLAGS="$CFLAGS -std=c99"
+CFLAGS="$CFLAGS -std=gnu99"
 # Do not warn about C++11 incompatibilities.
 CXXFLAGS="$CXXFLAGS -Wno-c++11-compat"
 
