--- origsrc/WebKit-r39090/GNUmakefile.am	2008-12-07 21:56:15.000000000 -0600
+++ src/WebKit-r39090/GNUmakefile.am	2008-12-07 21:56:00.000000000 -0600
@@ -75,6 +75,9 @@
 if OS_WIN32
 no_undefined = -no-undefined
 endif
+if OS_CYGWIN
+no_undefined = -no-undefined
+endif
 
 if OS_GNU
 version_script = -Wl,--version-script,$(srcdir)/symbols.filter
--- origsrc/WebKit-r39090/WebKitTools/GNUmakefile.am	2008-12-04 18:05:34.000000000 -0600
+++ src/WebKit-r39090/WebKitTools/GNUmakefile.am	2008-12-11 10:15:25.034250000 -0600
@@ -93,6 +95,9 @@
 	WebKitTools/DumpRenderTree/TestNetscapePlugIn.subproj/TestObject.cpp \
 	WebKitTools/DumpRenderTree/TestNetscapePlugIn.subproj/TestObject.h
 
+TestNetscapePlugin_libtestnetscapeplugin_la_LDFLAGS = \
+	-module -avoid-version $(no_undefined)
+
 # clean target
 CLEANFILES += \
 	Programs/GtkLauncher \
--- origsrc/WebKit-r39090/configure.ac	2008-12-07 21:56:16.000000000 -0600
+++ src/WebKit-r39090/configure.ac	2008-12-07 21:56:00.000000000 -0600
@@ -48,6 +48,9 @@
      *-*-darwin*)
        os_darwin=yes
        ;;
+     *-*-cygwin*)
+       os_cygwin=yes
+       ;;
 esac
 
 case "$host_os" in
@@ -704,6 +707,7 @@
 AM_CONDITIONAL([OS_LINUX],[test "$os_linux" = "yes"])
 AM_CONDITIONAL([OS_GNU],[test "$os_gnu" = "yes"])
 AM_CONDITIONAL([OS_FREEBSD],[test "$os_freebsd" = "yes"])
+AM_CONDITIONAL([OS_CYGWIN],[test "$os_cygwin" = "yes"])
 
 # target conditionals
 AM_CONDITIONAL([TARGET_X11], [test "$with_target" = "x11"])
